pkg/chartutil/values.go

```diff
@@ -142,39 +145,42 @@ func ReadValuesFile(filename string) (Values, error) {
 //		the values destined for its dependencies.
 func CoalesceValues(chrt *chart.Chart, vals *chart.Config) (Values, error) {
 	cvals := Values{}
+	// Default to using values unless otherwise noted by the vals config
+	confType := chart.ConfType_VALUES
```

Comment:
So if vals is nil, we default to `confType == VALUES`, otherwise we are able to get the config type from the vals object, which could be VALUES or SECRETS